### ISSUE TITLE
`blocks_in_if_conditions` shouldn't be triggered by a closure

### ISSUE THREAD

**Issue by c410-f3r** [rust-lang/rust-clippy#8099]
### Summary

Someway somehow a function closure parameter that returns `{}` triggers `blocks_in_if_conditions` if the function is wrapped inside an `if` condition.

### Reproducer

```rust
fn foo(_: impl FnMut()) -> bool {
    true
}

fn main() {
    // Triggers
    if foo(|| {}) {}

    // Does not trigger
    foo(|| {});
}
```

### Version

```text
rustc 1.59.0-nightly (0b6f079e4 2021-12-07)
binary: rustc
commit-hash: 0b6f079e4987ded15c13a15b734e7cfb8176839f
commit-date: 2021-12-07
host: x86_64-unknown-linux-gnu
release: 1.59.0-nightly
LLVM version: 13.0.0
```
